File: sdk/dart-jaguar/lib/api/product_conversion_api.dart
import 'package:jaguar_retrofit/annotations/annotations.dart';
import 'package:jaguar_retrofit/jaguar_retrofit.dart';
import 'package:jaguar_serializer/jaguar_serializer.dart';
import 'package:jaguar_mimetype/jaguar_mimetype.dart';
import 'dart:async';

import 'package:openapi/model/inline_response400.dart';
import 'package:openapi/model/inline_response200.dart';

part 'product_conversion_api.jretro.dart';

@GenApiClient()
class ProductConversionApi extends ApiClient with _$ProductConversionApiClient {
    final Route base;
    final Map<String, CodecRepo> converters;
    final Duration timeout;

    ProductConversionApi({this.base, this.converters, this.timeout = const Duration(minutes: 2)});

    /// Convert between UPC, EAN, and ASIN product codes.
    ///
    /// Returns the converted UPC, EAN, and ASIN codes.
    @GetReq(path: "/api/v1/convert/:upc | ean | asin")
    Future<InlineResponse200> convertCode(
            @PathParam("upc | ean | asin") String upc | ean | asin
        ) {
        return super.convertCode(
        upc | ean | asin

        ).timeout(timeout);
    }


}

File: sdk/scala-play-server/app/api/ProductConversionApi.scala
package api

import model.InlineResponse200
import model.InlineResponse400

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2020-03-27T18:11:16.680-04:00[America/New_York]")
trait ProductConversionApi {
  /**
    * Convert between UPC, EAN, and ASIN product codes.
    * Returns the converted UPC, EAN, and ASIN codes.
    * @param upc Pipe ean Pipe asin UPC, EAN, or ASIN
    */
  def convertCode(upc Pipe ean Pipe asin: String): InlineResponse200
}
